namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -225,4 +257,101 @@ class HttpControlService(storage: DtabStore) extends Service[Request, Response]
       case Throw(_: DtabNamespaceDoesNotExistException) => Future.value(Response(Status.NotFound))
       case Throw(_) => Future.value(Response(Status.InternalServerError))
     }
+
+  private[this] def streamingResp[T](values: Event[T], render: (T, Closable) => Buf): Future[Response] = {
+    val resp = Response()
+    resp.setChunked(true)
+    val writer = resp.writer
+    var closable: Closable = null
+    var writeFuture: Future[Unit] = Future.Unit
+    closable = values.respond { t =>
+      writeFuture = writeFuture.before {
+        writer.write(render(t, closable)).onFailure { _ =>
+          closable.close()
```

Comment:
I find this code a little difficult to understand, so maybe some comments would be in order. Are we guaranteed that `closable` will be non-`null` by the time this line of code is called? Why is writeFuture needed?